zebra-network/src/peer_set/initialize.rs

```diff
@@ -272,7 +272,10 @@ where
     let mut crawl_timer = tokio::time::interval(new_peer_interval);
 
     loop {
-        metrics::gauge!("crawler.in_flight_handshakes", handshakes.len() as i64 - 1);
+        metrics::gauge!(
+            "crawler.in_flight_handshakes",
+            (handshakes.len() - 1) as f64
```

Comment:
Do we need to check that this calculation is valid?
```suggestion
            handshakes.len().checked_sub(1).expect("the pool always contains an unresolved future") as f64
```